### PR TITLE
Implement `api/challenge` endpoints

### PR DESCRIPTION
**Pull request by Anupya**
## Description

Add all the remaining `challenge` endpoints:
- `get_mine` = `/api/challenge`
- `cancel` = `/api/challenge/{challengeId}/cancel`
- `start_clocks` = `/api/challenge/{gameId}/start-clocks`
- `add_time_to_opponent_clock` = `/api/round/{game_id}/add-time/{seconds}`
- `create_tokens_for_multiple_users` = `/api/token/admin-challenge`

Move `ChallengeDeclineReason` to `berserk/types/challenges.py` as `DeclineReason`.

## After merge

Please check off all `api/challenge/...` endpoints in the issue description for #6.

### CODE REVIEW

File: berserk/clients/challenges.py [hunk 1]
```diff
@@ -173,3 +181,56 @@ def decline(
...
+    ) -> None:
+        """Starts the clocks of a game immediately, even if a player has not yet made a move.
+
+        Requires the OAuth tokens of both players with challenge:write scope.
+
+        If the clocks have already started, the call will have no effect.
+
+        :param game_id: game ID
+        :param token_player_1: OAuth token of player 1 with challenge:write scope
+        :param token_player_2: OAuth token of player 2 with challenge:write scope
```
Comment (benediktwerner) [inline berserk/clients/challenges.py:207]: The docs here kinda suggest there is a specific order to the players (e.g. maybe player1 == white) but I'm pretty sure there isn't. It just has to be one token from white and one from black, in either order. I'd maybe just add a sentence or two explaining that.

**Comment by benediktwerner** [comment]
As mentioned in the other PR, return type annotations should ideally use TypedDict.

File: berserk/types/challenges.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    "created",
+    "offline",
+    "canceled",
+    "declined",
+    "accepted",
+]
+
+
+Title: TypeAlias = Literal[
+    "GM",
```
Comment (kraktus) [inline berserk/types/challenges.py:16]: You should use the ones defined in `.common.py` for Title, Colors etc, instead of redefining them here

**Comment by kraktus** [comment]
Perfect thanks! Just a NIT: https://github.com/lichess-org/berserk/pull/52/commits/ad22b757c0e613782c7f9866fb518eda043e8d1e

**Pull request merged**